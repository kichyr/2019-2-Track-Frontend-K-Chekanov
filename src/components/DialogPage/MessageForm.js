import React, { useEffect, useState } from 'react'
import { useHistory } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import ReactAudioPlayer from 'react-audio-player'
import PropTypes from 'prop-types'
import { usePosition } from 'use-position'
import singleMessStyles from './singleMessage.module.css'
import WHOAMI from '../whoami'
import topLineStyles from './topLine.module.css'
import styles from './styles.module.css'
import BackArrow from '../BackArrow/BackArrow'
import { PAPER_AIRPLANE, CLIP, FILE_ICON } from './svgVariables'
import { openAudioMessagePage } from '../../actions/sendMessage'
import { AudioMessageSender, sendFile, Message } from './utils'

/* function getChatMessages(chatId) {
	const messages = localStorage.getItem('messages');
	return [ messages[chatId].topic, messages[chatId].messages ];
} */

function postMessage(message) {
  const data = JSON.parse(localStorage.getItem('messages'))
  data[message.chatId].push(message)
  localStorage.setItem('messages', JSON.stringify(data))
}

function TopLine({ topic, appState, setAppState }) {
  const history = useHistory()
  return (
    <div id="chatTopLine" className={topLineStyles.container}>
      <div
        id="chatBack"
        onClick={(e) => {
          setAppState({ ...appState, appPage: 'ChatList', prevAppPage: history.location })
          history.push(`${window.publicUrl}/`)
        }}
        onKeyDown={(e) => {}}
        style={{ flex: '0.2' }}
        role="button"
        tabIndex={0}
      >
        <BackArrow />
      </div>
      <div className={topLineStyles.topic}> {topic} </div>
    </div>
  )
}

function MessagesContainer({ messages, appState, setAppState }) {
  const [dragging, setDragging] = useState(false)
  const mContainer = React.createRef()
  const history = useHistory()
  // eslint-disable-next-line
  useEffect(() => {
    // Обновляем заголовок документа с помощью API браузера
    // eslint-disable-next-line
    mContainer.current.scrollTop = mContainer.current.scrollHeight
    // eslint-disable-next-line
  }, [messages])

  // DRAGDandDROP
  const handleDragIn = (e) => {
    e.preventDefault()
    e.stopPropagation()
    setDragging(true)
  }

  const handleDragOut = (e) => {
    e.persist()
    e.preventDefault()
    e.stopPropagation()
    setDragging(false)
  }

  const handleDrop = (e) => {
    setDragging(false)
    e.preventDefault()
    e.stopPropagation()
    if (e.dataTransfer.items[0].kind !== 'file') return
    const file = e.dataTransfer.items[0].getAsFile()
    sendFile(file, appState, setAppState)
  }
  // END DRAGDandDROP
  return (
    <div className={styles.result} style={{ scrollTop: 'scrollHeight' }} ref={mContainer} onDragEnter={handleDragIn}>
      <DragAndDropImg dragging={dragging} handleDragOut={handleDragOut} handleDrop={handleDrop} />
      <AudioMessageSender appState={appState} setAppState={setAppState} />

      {messages.map((message, index) => (
        <div
          key={index.toString()}
          className={singleMessStyles.singleMessContainer}
          onDrop={(e) => {
            e.preventDefault()
          }}
        >
          <input
            type="image"
            onClick={(e) => {
              // setAppState(Object.assign({}, appState, {appPage: 'ProfilePage', prevAppPage: 'Chat'}));
              history.push(`${window.publicUrl}/profile`, appState)
            }}
            className={topLineStyles.chatImg}
            src="http://emilcarlsson.se/assets/rachelzane.png"
            alt="Avatar"
          />
          <div className={singleMessStyles.singleMess} sender={message.userId === WHOAMI.userId ? 'me' : 'him'}>
            <div className={singleMessStyles.singleMessText}>
              {(() => {
                switch (message.contentType) {
                  case 'link':
                    return <a href={message.messageText}> {message.messageText} </a>
                  case 'image':
                    return <img alt="" src={message.link} style={{ maxWidth: '100%' }} />
                  case 'file':
                    return (
                      <div style={{ display: 'inline' }}>
                        <a
                          aria-label="location"
                          // eslint-disable-next-line
                          dangerouslySetInnerHTML={{ __html: FILE_ICON }}
                          href={message.link}
                          style={{
                            height: '50%',
                          }}
                        />
                        <p>{message.fileName}</p>
                      </div>
                    )
                  case 'audioMessage':
                    return (
                      <div>
                        <ReactAudioPlayer src={message.link} autoPlay controls />
                      </div>
                    )
                  default:
                    return <div> {message.messageText} </div>
                }
              })()}
            </div>
            <p className={singleMessStyles.dateTime}> {message.time} </p>
          </div>
        </div>
      ))}
    </div>
  )
}

function InputPanel({ appState, setAppState }) {
  const dispatch = useDispatch()

  const sendMessage = (e) => {
    e.preventDefault()
    const messText = e.target[0].value
    if (messText.replace(/\s/gi, '') === '') {
      return
    }
    const newMess = new Message(appState.openedChat.chatId, WHOAMI.userId, messText, 'text')
    postMessage(newMess)
    setAppState({
      ...appState,
      ...Object.assign(appState.openedChat, { messages: [...appState.openedChat.messages, newMess] }),
    })
    e.target[0].value = ''
  }

  const textInput = React.createRef()
  const attachmentTypeDiv = React.createRef()
  const { latitude, longitude } = usePosition()

  const GeolocationHandler = (e) => {
    e.preventDefault()
    const newMess = new Message(
      appState.openedChat.chatId,
      WHOAMI.userId,
      `https://www.openstreetmap.org/#map=18/${latitude}/${longitude}`,
      'link',
    )
    postMessage(newMess)
    setAppState({
      ...appState,
      ...Object.assign(appState.openedChat, { messages: [...appState.openedChat.messages, newMess] }),
    })
    attachmentTypeDiv.current.style.transform = 'scale(0)'
  }

  return (
    <div className={styles.input_panel}>
      <form className={styles.sendMessForm} onSubmit={sendMessage}>
        <input
          ref={textInput}
          type="text"
          className={styles.messageText}
          maxLength="512"
          placeholder="Введите сообщеине"
        />
        <div
          style={{
            margin: '5px',
            flex: '0.3',
            maxHeight: '100%',
          }}
        >
          <div ref={attachmentTypeDiv} className={styles.attachmentType}>
            <label id="input">
              <input
                type="file"
                name="file"
                onChange={(e) => {
                  attachmentTypeDiv.current.style.transform = 'scale(0)'
                  sendFile(e.target.files[0], appState, setAppState)
                }}
                onKeyDown={(e) => {}}
                style={{ display: 'none' }}
              />
              <div>Файл</div>
            </label>
            <div role="button" onClick={GeolocationHandler} onKeyDown={(e) => {}}>
              Моя геолокации
            </div>
            <div
              onClick={(e) => {
                dispatch(openAudioMessagePage())
                attachmentTypeDiv.current.style.transform = 'scale(0)'
              }}
            >
              {' '}
              Аудиосообщение{' '}
            </div>
          </div>
          <div
            aria-label="send message"
            // eslint-disable-next-line
            dangerouslySetInnerHTML={{ __html: CLIP }}
            style={{
              height: '100%',
            }}
            role="button"
            tabIndex={-1}
            onKeyDown={() => {}}
            onClick={(e) => {
              attachmentTypeDiv.current.style.transform =
                attachmentTypeDiv.current.style.transform !== 'scale(1)' ? 'scale(1)' : 'scale(0)'
            }}
          />
        </div>

        <label
          style={{
            flex: '0.3',
            display: 'flex',
            maxHeight: '100%',
          }}
        >
          <input style={{ display: 'none' }} type="submit" value="" />

          <div dangerouslySetInnerHTML={{ __html: PAPER_AIRPLANE }} style={{ flex: '1', margin: '5px' }} />
        </label>
      </form>
    </div>
  )
}

function MessageForm({ appState, setAppState }) {
  return (
    <div className={styles.message_form}>
      <TopLine topic={appState.openedChat.topic} appState={appState} setAppState={setAppState} />
      <MessagesContainer messages={appState.openedChat.messages} appState={appState} setAppState={setAppState} />
      <InputPanel appState={appState} setAppState={setAppState} />
    </div>
  )
}

function DragAndDropImg({ dragging, handleDragOut, handleDrop }) {
  return (
    dragging && (
      <div
        onDragLeave={handleDragOut}
        onDrop={handleDrop}
        onDragOver={(e) => {
          e.preventDefault()
          e.stopPropagation()
        }}
        onDragEnter={(e) => {
          e.preventDefault()
          e.stopPropagation()
        }}
        style={{
          border: 'dashed grey 4px',
          backgroundColor: 'rgba(255,255,255,.8)',
          position: 'absolute',
          top: '10vh',
          bottom: '10vh',
          left: 0,
          right: 0,
          zIndex: 9999,
        }}
      >
        <div
          style={{
            position: 'absolute',
            top: '50%',
            right: 0,
            left: 0,
            textAlign: 'center',
            color: 'grey',
            fontSize: 36,
          }}
        >
          <div>drop here :)</div>
        </div>
      </div>
    )
  )
}

MessageForm.propTypes = {
  appState: PropTypes.shape({
    appPage: PropTypes.string.isRequired,
    openedChat: PropTypes.shape({
      chatId: PropTypes.number,
      topic: PropTypes.string.isRequired,
      messages: PropTypes.array.isRequired,
    }),
  }).isRequired,
  setAppState: PropTypes.func.isRequired,
}

InputPanel.propTypes = {
  appState: PropTypes.shape({
    appPage: PropTypes.string.isRequired,
    openedChat: PropTypes.shape({
      chatId: PropTypes.number,
      topic: PropTypes.string.isRequired,
      messages: PropTypes.array.isRequired,
    }),
  }).isRequired,
  setAppState: PropTypes.func.isRequired,
}

MessagesContainer.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.object).isRequired,
  appState: PropTypes.shape({
    appPage: PropTypes.string.isRequired,
    openedChat: PropTypes.shape({
      chatId: PropTypes.number,
      topic: PropTypes.string.isRequired,
      messages: PropTypes.array.isRequired,
    }),
  }).isRequired,
  setAppState: PropTypes.func.isRequired,
}

TopLine.propTypes = {
  topic: PropTypes.string.isRequired,
  appState: PropTypes.shape({
    appPage: PropTypes.string.isRequired,
    openedChat: PropTypes.shape({
      chatId: PropTypes.number,
      topic: PropTypes.string.isRequired,
      messages: PropTypes.array.isRequired,
    }),
  }).isRequired,
  setAppState: PropTypes.func.isRequired,
}

export default MessageForm
