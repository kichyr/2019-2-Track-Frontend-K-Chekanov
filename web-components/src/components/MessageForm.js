import {Message, saveMess} from "./LocalStorage"
const template = document.createElement('template');
template.innerHTML = `
    <style>
        top-line {
            flex: 1.5;
        }
        .input_panel {
            background: #eee;
            flex: 1;
            padding: 25px;
            opacity: 1;
            display: flex;
            font-family: Helvetica, Arial, sans-serif;
        }

        textarea {
            flex: 8;
            font-size: 3vh;
            border-radius: 20px;
            overflow:hidden;
            padding-left: 20px;
        }
        button {
            background-color: #0084ff;
            color: #fff;
            font-size: 3vh;
            border-radius: 20px;
            flex: 1;
        }

        form {
            flex: 1;
            height: 100%;
            display: flex;
            flex-direction: column;
        }

        .result {
            vertical-align: bottom;
            color: blue;
            overflow-y: auto;
            flex: 14;
        }
        ::-webkit-scrollbar {
            width: 12px;
            height: 12px;
        }
            
        ::-webkit-scrollbar-track {
            border: 1px solid blue;
            border-radius: 10px;
            }
            
        ::-webkit-scrollbar-thumb {
            background: blue;
            border-radius: 10px;
            }
            
        ::-webkit-scrollbar-thumb:hover {
            background: #88ba1c;  
            }

        input[type=submit] {
            visibility: collapse;
        }
        
          
        .singleMess {
            display:inline-block;
            clear: both;
            padding: 20px;
            border-radius: 30px;
            margin-bottom: 2px;
            font-size: 3vh;
            max-width: 70%; 
            margin-top: 10px;
            text-align:right;
        }
        
        [sender = him]{
        background: #eee;
        float: left;
        }
        
        [sender = me]{
        float: right;
        background: #0084ff;
        color: #fff;
        }
        
    [sender]{
        border-bottom-right-radius: 5px;
    }
        
    [sender = me]{
        border-top-right-radius: 5px;
        border-bottom-right-radius: 5px;
    }
        
    [sender = him]:last-of-type {
        border-bottom-right-radius: 30px;
    }

    .dateTime {
        font-size: 2vh;
        margin: 0;
        align: right;
    }
    [sender = me] .dateTime {
        color: LightGray;
    }
    [sender = him] .dateTime {
        color: Gray;
    }
        
    </style>
    <form>
        <top-line></top-line>
        <div class="result"></div>
        <div class = input_panel>
            <textarea name="message-text" placeholder="Введите сообщеине"></textarea>
            <button>Send</button>
        </div>
    </form>`;


var WHOAMI = 'me';

class MessageForm extends HTMLElement {
    constructor () {
        super();
        this.interlocutorName = "him";
        this._shadowRoot = this.attachShadow({ mode: 'open' });
        this._shadowRoot.appendChild(template.content.cloneNode(true));
        this.$form = this._shadowRoot.querySelector('form');
        this.$input = this._shadowRoot.querySelector('textarea');
        this.$messages = this._shadowRoot.querySelector('.result');
        this.loadOldMessages();

        this.$form.addEventListener('submit', this._onSubmit.bind(this));
        this.$form.addEventListener('keypress', this._onKeyPress.bind(this));
    }

    loadOldMessages() {
        if(localStorage.getItem('messeges') == "")
            return;
        var data = JSON.parse(localStorage.getItem('messeges'));
        for (let i = 0; i < data.length; i++) {
            this.displayMessage(data[i])
        }
    }

    _onSubmit (event) {
        event.preventDefault();
        var message = new Message(this.$input.value, WHOAMI);
        saveMess(message);
        this.displayMessage(message);
        //---------------TO_FIX-----------------
        var message_bot = new Message("Hi, fix me later pls)", this.interlocutorName);
        saveMess(message_bot);
        this.displayMessage(message_bot);
        // -------------------------------------
        this.scrollToBot();
        this.$input.value = '';
    }

    dateTime() {
        var today = new Date();
        var time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
        var today = new Date();
        var date = today.getFullYear()+'-'+(today.getMonth()+1)+'-'+today.getDate();
        return date + " " + time;
    }

    _onKeyPress(event) {
        if (event.keyCode == 13) {
            this.$form.dispatchEvent(new Event('submit'));
        }
    }

    scrollToBot() {
        let chatHistory = this.$messages;
        chatHistory.scrollTop = chatHistory.scrollHeight;
    }

    displayMessage(message) {
        var mess = document.createElement('div');
        mess.setAttribute("class", "singleMess");
        var dateTime = document.createElement('p');
        dateTime.setAttribute("class", "dateTime");
        dateTime.innerHTML = message.time;
        if(message.sender == WHOAMI)
            mess.setAttribute("sender", "me");
        else
            mess.setAttribute("sender", "him");
        mess.innerHTML = message.messageText;
        mess.appendChild(dateTime);
        this.$messages.appendChild(mess);
    }
}

customElements.define('message-form', MessageForm);
