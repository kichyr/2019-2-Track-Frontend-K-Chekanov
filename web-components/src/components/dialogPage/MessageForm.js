import {Message, saveMess} from "./LocalStorage"

const template = document.createElement('template');
template.innerHTML = `
    <style>
        top-line {
            flex: 1.5;

        }
        
        .input_panel {
            background: #eee;
            flex: 1;
            padding: 25px;
            opacity: 1;
            display: flex;
            font-family: Helvetica, Arial, sans-serif;
        }

        textarea {
            flex: 8;
            color: transparent;
            text-shadow: 0px 0px 0px #000000;
            font-size: 3vh;
            border-radius: 20px;
            overflow:hidden;
            padding-left: 20px;
        }
        button {
            background-color: #0084ff;
            color: #fff;
            font-size: 3vh;
            border-radius: 20px;
            flex: 1;
        }

        form {
            flex: 1;
            height: 100%;
            display: flex;
            flex-direction: column;
        }

        .result {
            vertical-align: bottom;
            color: blue;
            overflow-y: auto;
            flex: 14;
        }
        ::-webkit-scrollbar {
            width: 12px;
            height: 12px;
        }
            
        ::-webkit-scrollbar-track {
            border: 1px solid blue;
            border-radius: 10px;
        }
            
        ::-webkit-scrollbar-thumb {
            background: blue;
            border-radius: 10px;
            }
            
        ::-webkit-scrollbar-thumb:hover {
            background: #88ba1c;  
            }

        input[type=submit] {
            visibility: collapse;
        }
    </style>
    <form>
        <top-line></top-line>
        <div class="result"></div>
        <div class = input_panel>
            <textarea name="message-text" placeholder="Введите сообщеине"></textarea>
            <button>Send</button>
        </div>
    </form>`;


var WHOAMI = 'me';

class MessageForm extends HTMLElement {
    constructor () {
        super();
        this.interlocutorName = "him";
        this._shadowRoot = this.attachShadow({ mode: 'open' });
        this._shadowRoot.appendChild(template.content.cloneNode(true));
        this.$form = this._shadowRoot.querySelector('form');
        this.$input = this._shadowRoot.querySelector('textarea');
        this.$messages = this._shadowRoot.querySelector('.result');
        this.scrollToBot();
        this.$form.addEventListener('submit', this._onSubmit.bind(this));
        //this.$form.addEventListener('keypress', this._onKeyPress.bind(this));
        console.log(localStorage.getItem('messages'));
    }

    loadOldMessages(login) {
        this.$messages.innerHTML = '';
        this.login = login;
        if(localStorage.getItem('messages') == null)
            return
        var data = JSON.parse(localStorage.getItem('messages'));
        console.log("kek");
        data = data[login];
        for (let i = 0; i < data.length; i++) {
            this.displayMessage(data[i])
        }
    }

    _onSubmit (event) {
        event.preventDefault();
        if(this.$input.value == "")
            return;
        var message = new Message(this.$input.value, WHOAMI);
        saveMess(message, this.login);
        this.displayMessage(message);
        //---------------TO_FIX-----------------
        var message_bot = new Message("Hi, fix me later pls)", this.interlocutorName);
        saveMess(message_bot, this.login);
        this.displayMessage(message_bot);
        // -------------------------------------
        this.scrollToBot();
        this.$input.value = '';
    }

    dateTime() {
        var today = new Date();
        var time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
        var date = today.getFullYear()+'-'+(today.getMonth()+1)+'-'+today.getDate();
        return date + " " + time;
    }

    _onKeyPress(event) {
        if (event.keyCode == 13) {
            this.$form.dispatchEvent(new Event('submit'));
        }
    }

    scrollToBot() {
        let chatHistory = this.$messages;
        chatHistory.scrollTop = chatHistory.scrollHeight;
    }

    displayMessage(message) {
        var mess = document.createElement('single-mess');
        mess.init(message, WHOAMI);
        this.$messages.appendChild(mess);
    }
}

customElements.define('message-form', MessageForm);