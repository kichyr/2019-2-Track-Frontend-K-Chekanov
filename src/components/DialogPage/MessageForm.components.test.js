import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

import { DragAndDropImg } from './MessageForm'

jest.mock('react-mic', () => ({
  ReactMic: () => ({
    push: jest.fn(),
  }),
}))

it('renders drag and drop background', () => {
  const output = shallow(<DragAndDropImg dragging={(e) => {}} handleDragOut={(e) => {}} handleDrop={(e) => {}} />)
  expect(output).toMatchSnapshot()
})
