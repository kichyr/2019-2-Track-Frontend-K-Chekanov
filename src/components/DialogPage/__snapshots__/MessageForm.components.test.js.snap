// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders drag and drop background 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DragAndDropImg
    dragging={[Function]}
    handleDragOut={[Function]}
    handleDrop={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
          Object {
            "color": "grey",
            "fontSize": 36,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "textAlign": "center",
            "top": "50%",
          }
        }
      >
        <div>
          drop here :)
        </div>
      </div>,
      "onDragEnter": [Function],
      "onDragLeave": [Function],
      "onDragOver": [Function],
      "onDrop": [Function],
      "style": Object {
        "backgroundColor": "rgba(255,255,255,.8)",
        "border": "dashed grey 4px",
        "bottom": "10vh",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": "10vh",
        "zIndex": 9999,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          drop here :)
        </div>,
        "style": Object {
          "color": "grey",
          "fontSize": 36,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "textAlign": "center",
          "top": "50%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "drop here :)",
        },
        "ref": null,
        "rendered": "drop here :)",
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
            Object {
              "color": "grey",
              "fontSize": 36,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "textAlign": "center",
              "top": "50%",
            }
          }
        >
          <div>
            drop here :)
          </div>
        </div>,
        "onDragEnter": [Function],
        "onDragLeave": [Function],
        "onDragOver": [Function],
        "onDrop": [Function],
        "style": Object {
          "backgroundColor": "rgba(255,255,255,.8)",
          "border": "dashed grey 4px",
          "bottom": "10vh",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": "10vh",
          "zIndex": 9999,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            drop here :)
          </div>,
          "style": Object {
            "color": "grey",
            "fontSize": 36,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "textAlign": "center",
            "top": "50%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "drop here :)",
          },
          "ref": null,
          "rendered": "drop here :)",
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
